data-processing/tests/test_extract_definitions.py

```diff
@@ -219,6 +366,41 @@ def test_extract_nickname_for_th_index_pattern():
     assert nickname0.term_text == "k"
     assert nickname0.definition_text == "timestep"
 
+@pytest.mark.slow
+def test_model_extract_abbreviation_acronym(model: DefinitionDetectionModel):
+    prediction_type = 'ABBR-EXP'
+    features = model.featurize("We use a Convolutional Neural Network (CNN) based architecture")
+    intents, slots, _ = model.predict_batch([features], prediction_type)
+    assert intents[0]
+    assert slots[0] == ["O", "O", "O", "DEF", "DEF", "DEF", "O", "TERM", "O", "O", "O"]
+
+
+@pytest.mark.slow
+def test_extract_abbreviation_expansion_pairs_case_1():
+    prediction_type = 'ABBR-EXP'
+    text = "We use a Convolutional Neural Network (CNN) based architecture"
+    tokens, slots, confidences = zip(
+        *[
+            ("We", "O", 1.0),
+            ("use", "O", 1.0),
+            ("a", "O", 1.0),
+            ("Convolutional", "DEF", 1.0),
+            ("Neural", "DEF", 1.0),
+            ("Network", "DEF", 1.0),
+            ("(", "O", 1.0),
+            ("CNN", "TERM", 1.0),
+            (")", "O", 1.0),
+            ("based", "O", 1.0),
+            ("architecture", "O", 1.0),
+        ]
+    )
+
+    abbreviation_expansion_pairs = consolidate_keyword_definitions(
+        text, tokens, slots, confidences, prediction_type
+    )
```

Comment:
@nuwandavek this part of the test feels more like a test of `consolidate_keyword_definitions` than it does of the model. In fact, the model is not even getting invoked here. Do we even need this test? Maybe just make a couple of dedicated tests for `consolidate_keyword_definitions` and call it a day?

Reply:
Yeah, that sounds good!